src/generators/dom/index.js

```diff
@@ -168,7 +169,7 @@ export default function dom ( parsed, source, options ) {
 	if ( templateProperties.oncreate ) {
 		builders.init.addBlock( deindent`
 			if ( options._root ) {
-				options._root._renderHooks.push({ fn: ${generator.alias( 'template' )}.oncreate, context: this });
+				options._root._renderHooks.push( ${generator.alias( 'template' )}.oncreate.bind( this ) );
```

Comment:
That's probably better, yeah. If we wanted to be really clever, the fastest way to run `oncreate` would probably be to rewrite references to `this`, the same way we do in non-hoisted event handlers:

```js
// this...
oncreate () {
  this.observe( 'foo', function ( foo ) {
    this.refs.whatever.value = foo.toUpperCase();
  });
}

// ...becomes this:
oncreate ( component ) {
  component.observe( 'foo', function ( foo ) {
    this.refs.whatever.value = foo.toUpperCase();
  });
}
```

Or is that a terrible idea?